### ISSUE TITLE
Fedilab returns public status of wrong account while trying to use it to log in to an account on nerdica.net

### ISSUE THREAD

**Issue by ChaoticHuman** [friendica/friendica#7311]
### Expected behavior
-Normal log in to the chosen nerdica.net-account
-nerdica.net asks to authorize the Fedilab-app

### Actual behavior
-Return of the wrong user-account but only it's public statuses
-nerdica.net doesn't ask to authorize the Fedilab-app

### Steps to reproduce the problem
1.Download, install and start Fedilab Version 2.6.1 from F-Droid.
2. Use nerdica.net as instance and connect
3. Use user-credentials of your nerdica.net-account and log in

### Friendica version you encountered the problem
-Friendica-version 2019.06-rc on the instance nerdica.net
-Fedilab-version 2.6.1

**Comment by ingoj** [comment]
From https://nerdica.net/display/22b2c6dfa971438ade7d8801c0136516b2661726 : 

> Not a security issue I confirm. I checked it deeply. As soon as it requires a token, it throws a 401.
> But, the reply for the random account comes from the server reply. 

So, no security issue as it seems, but apparently a misleading behavior of Friendica that can be solved by either changing the code or changing .htaccess using a redirect to the main page or an extra error page.

**Comment by stom79** [comment]
When using this endpoint with nerdica.net: `/api/account/verify_credentials.json` it returns a user object that has nothing to do with the one specified via Basic auth. 
I only tested with fake credentials (test:test) as I don't have an account. No data are leaked, because the token is not correct but the app displays the account object returned with the endpoint.

**Comment by MrPetovan** [comment]
Are there any query string parameters (`?key=value&...`) sent to this endpoint?

**Comment by stom79** [comment]
I checked and only `?`  remains at the end of the URL.

**Comment by stom79** [comment]
I can't reproduce that bug now, even by removing or not the question mark at the end. I guess it's fixed?

**Comment by MrPetovan** [comment]
Without query string parameters, `verify_credentials` returns the info of the user currently logged in in session. I'm not sure what to make of this because getting the wrong user info would mean the underlying session feature is broken, which is both terrifying and extremely unlikely.

**Comment by stom79** [comment]
I don't clearly understand why adding a mark point to the end of the URL can render a user that doesn't match credentials. 
Now, I can't get this issue back but I don't know if something has been done.
Have you some assumptions about this issue?

**Comment by MrPetovan** [comment]
Having just a question mark is the same as having none, but the code reacts differently depending to a number of query string parameters, hence my question. Your answer meant that the user info is coming from the logged in session instead.

**Comment by stom79** [comment]
No, because with Fedilab I never used nerdica.net and I tested with test as username and test as password and I got an account in reply.
Of course the markdow at the end of a URL doesn't change nothing, but I wonder when this call was caught from the server, a piece of code could be executed where it should not.

**Comment by MrPetovan** [comment]
It still is mysterious, granted, but at least we ruled out query string parameters.

**Comment by ingoj** [comment]
Well, I assume that it is a general issue, if at all, and not an issue related to the setup of Nerdica.
